pkg/apis/core/types.go

```diff
@@ -2138,31 +2138,31 @@ const (
 	PullIfNotPresent PullPolicy = "IfNotPresent"
 )
 
-// ResourceResizePolicy specifies how Kubernetes should handle resource resize.
-type ResourceResizePolicy string
+// ResourceResizeRestartPolicy specifies how to handle container resource resize.
+type ResourceResizeRestartPolicy string
```

Comment:
Would the name `ResourceResizeApplyPolicy` be a better name? What if it will be something beyond restart at some point?

Reply:
it comes to the @thockin comment later. `Apply` instead of `Restart` may be used later to make Pod level policies or applying with special signal sent to container.

Reply:
The current model is closer to what you are describing.  The reason I asked for this sort of change was that I think other policies are plausible and having a single `policy` field means you get one "namespace" of policies.  I think multiple ortogonal xyzPolicy fields makes more sense.

Some discussion here: https://github.com/kubernetes/kubernetes/pull/111946#discussion_r1068797501

Reply:
I see. Would making it in a struct be desirable than?

Is your comment regarding `Container` vs. `Pod` an example of another dimension of a policy? Or it is clarification of a restart policy?

Reply:
I think "restart pod" would be another value to this "restart policy".  I don't have an obvious "other policy" that feels realistic, but suppose we wanted some sort of notification policy or something like that, you'd want a `notificationPolicy` field, right?  Just because I can't think of one doesn't mean that a really great idea won't emerge.  If anything, I think I have just shown that my imagination is very limited :)

I don't think we need a struct here, though.  It's not a one-of or anything, I and think:

```
policy:
    restart: NotRequired
```

is not better (and probably worse) than:

```
restart Policy: NotRequired
```

Reply:
yep. Agree with all the reasoning. Struct is premature and not introducing much value. RestartPod will be great for the future.